src/OverlayTrigger.js

```diff
@@ -77,192 +100,168 @@ const propTypes = {
 };
 
 const defaultProps = {
-  defaultOverlayShown: false,
+  defaultShow: false,
   trigger: ['hover', 'focus'],
 };
 
-// eslint-disable-next-line react/no-multi-comp
-class OverlayTrigger extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-
-    this.trigger = React.createRef();
-    this.state = {
-      show: !!props.defaultShow,
+function OverlayTrigger({
+  trigger,
+  overlay,
+  children,
+  popperConfig = {},
+  defaultShow,
+  delay: propsDelay,
+  ...props
+}) {
+  const triggerNodeRef = useRef(null);
+  const timeoutRef = useRef();
+  const hoverStateRef = useRef();
+  const [show, setShow] = useState(!!defaultShow);
+
+  useEffect(() => {
+    return () => {
+      clearTimeout(timeoutRef.current);
     };
+  }, []);
 
-    // We add aria-describedby in the case where the overlay is a role="tooltip"
-    // for other cases describedby isn't appropriate (e.g. a popover with inputs) so we don't add it.
-    this.ariaModifier = {
-      enabled: true,
-      order: 900,
-      fn: data => {
-        const { popper } = data.instance;
-        const target = this.getTarget();
-        if (!this.state.show || !target) return data;
-
-        const role = popper.getAttribute('role') || '';
-        if (popper.id && role.toLowerCase() === 'tooltip') {
-          target.setAttribute('aria-describedby', popper.id);
-        }
-        return data;
-      },
-    };
-  }
+  const delay = normalizeDelay(propsDelay);
 
-  componentWillUnmount() {
-    clearTimeout(this._timeout);
-  }
+  const child = React.Children.only(children);
+  const { onFocus, onBlur, onClick } = child.props;
 
-  getChildProps() {
-    return React.Children.only(this.props.children).props;
-  }
+  const getTarget = useCallback(
+    () => ReactDOM.findDOMNode(triggerNodeRef.current),
```

Comment:
we should consider dropping the `findDOMNode` before v1 like we did for dropdowns

Reply:
@jquense would that be out of scope for this PR? Or should we do this in this PR?

Reply:
not in scope for this PR, i don't think